src/gc.c

```diff
@@ -1,12 +1,78 @@
 // This file is a part of Julia. License is MIT: https://julialang.org/license
 
 #include "gc.h"
+#include "julia_gcext.h"
 #include "julia_assert.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+// Linked list of callback functions
+
+typedef void (*jl_gc_cb_func_t)(void);
+
+typedef struct jl_gc_callback_list_t {
+    struct jl_gc_callback_list_t *next;
+    jl_gc_cb_func_t func;
+} jl_gc_callback_list_t;
+
+static jl_gc_callback_list_t *gc_cblist_pre_gc;
+static jl_gc_callback_list_t *gc_cblist_post_gc;
+
+#define gc_invoke_callbacks(kind, args) \
```

Comment:
The macro symbol concatenation is still here.....

Reply:
So, just to clarify: you prefer that we replicate this code 6 times? The justification being that this way one can grep for e.g. `gc_cblist_root_scanner` to find where it is used?

A simple alternative to achieve that would be to add a comment to each of the 6 invocations of `gc_invoke_callbacks`, e.g.
```C
    // iterate over gc_cblist_root_scanner
    gc_invoke_callbacks(notify_external_alloc, (v, allocsz));
```